ci/scripts/install_gcs_testbench.sh

```diff
@@ -34,19 +34,23 @@ case "$(uname -m)" in
     ;;
 esac
 
-# On newer pythons install into the system will fail, so override that
-export PIP_BREAK_SYSTEM_PACKAGES=1
-
 version=$1
 if [[ "${version}" -eq "default" ]]; then
   version="v0.39.0"
-  # Latests versions of Testbench require newer setuptools
-  python3 -m pip install --upgrade setuptools
 fi
 
+: ${PIPX_PYTHON:=$(which python3)}
+
+export PIP_BREAK_SYSTEM_PACKAGES=1
+${PIPX_PYTHON} -m pip install -U pipx
+
 # This script is run with PYTHON undefined in some places,
 # but those only use older pythons.
 if [[ -z "${PYTHON_VERSION}" ]] || [[ "${PYTHON_VERSION}" != "3.13" ]]; then
```

Comment:
This if check could be removed now? (AFAIK we only added it because we couldn't install grpcio on python 3.13, but now this is using pipx (with fixed python version) it shouldn't matter there are no python 3.13 wheels yet?

Reply:
Yes, I should open a followup issue.

Reply:
Opened https://github.com/apache/arrow/issues/43883